models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -920,6 +920,10 @@ public function checkValidity(mixed $data, bool $omitMandatoryCheck = false, arr
                             $data = $blockElement->getData();
 
                             if ($data instanceof DataObject\Localizedfield && $fd instanceof Localizedfields) {
+
+                                if (isset($params['object'])){
```

Comment:
Is this the right place to set Object? or should we set on load?

Reply:
For the purpose, it seemed the best place there, not sure if it could be passed elsewhere/earlier.
Where do you mean for `on load`, is there an example?

Reply:
I mean `checkValidity` is called on update and the object references should be renewed on load, just like we do it for fieldcollections, objectbrick and localizedfields https://github.com/pimcore/pimcore/blob/a8d0258495aa3d5b6a25f95ae2b4f3b0d0abc66d/models/DataObject/Concrete.php#L774

Reply:
preSetData works fine, thank you for the suggestion